File: pyg_parent/pyg_web_user/src/main/webapp/js/service/userService.js
//用户服务层
app.service('userService',function($http){
    this.add= function(user,smscode){
        return $http.post( 'user/add.action?smscode='+smscode,user);
    }

    //发送短信验证码
    this.sendCode= function(phone){
        return $http.get( 'user/sendCode.action?phone='+phone);
    }

});

File: pyg_parent/pyg_interface/src/main/java/com/itheima/pyg/service/user/UserService.java
package com.itheima.pyg.service.user;

import com.itheima.pyg.entity.PageResult;
import com.itheima.pyg.pojo.user.User;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.Map;

public interface UserService {
    void addUser(User user);

    void sendCode(String phone);

    boolean checkSmsCode(String phone,String smscode);

    List<User>  getUserList();

    PageResult<User> getUserListByPage(Integer pageNum, Integer pageSize);

    void save(User user);


}
